repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/valkyrie/ValkyrieAuthorizationQualityTest.groovy

```diff
@@ -0,0 +1,176 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.filters.valkyrie
+
+import org.openrepose.framework.test.ReposeValveTest
+import org.openrepose.framework.test.mocks.MockValkyrie
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+import spock.lang.Shared
+import spock.lang.Unroll
+
+import static javax.servlet.http.HttpServletResponse.SC_FORBIDDEN
+import static javax.servlet.http.HttpServletResponse.SC_OK
+import static org.openrepose.commons.utils.http.CommonHttpHeader.AUTH_TOKEN
+import static org.openrepose.commons.utils.http.OpenStackServiceHeader.CONTACT_ID
+import static org.openrepose.commons.utils.http.OpenStackServiceHeader.TENANT_ID
+
+class ValkyrieAuthorizationQualityTest extends ReposeValveTest {
+    static def HDR_DEVICE_ID = "X-Device-Id"
+
+    @Shared
+    MockValkyrie mockValkyrie
+
+    def setupSpec() {
+        Map params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/valkyrie/quality", params)
+
+        repose.start()
+
+        mockValkyrie = new MockValkyrie(properties.valkyriePort)
+
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(port: properties.targetPort)
+        deproxy.addEndpoint(port: properties.valkyriePort, name: "valkyrie service", defaultHandler: mockValkyrie.getHandler())
+    }
+
+    def setup() {
+        mockValkyrie.resetCounts()
+        mockValkyrie.resetHandlers()
+        mockValkyrie.resetParameters()
```

Comment:
Could you set the `contact_id` to some new, unique value here instead of in the tests? It seems like it's just being used to avoid cache collisions anyway, since the mock will, by default, return a body only containing permissions for the `contact_id` it pulls from the URL.

Reply:
Ideally, to keep things isolated, the `contact_id` would not be the same as the `tenant_id`. Not that it really matters, it just lead to me asking why that was always the case, so I'm thinking more isolation might make it more obvious that those values do not have to match.